File: chrome/test/data/webui/app_home/app_home_browsertest.js
// Copyright 2022 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/** @fileoverview Test suite for the WebUI app home. */
GEN_INCLUDE(['//chrome/test/data/webui/polymer_browser_test_base.js']);

GEN('#include "content/public/test/browser_test.h"');

GEN('#include "chrome/browser/ui/ui_features.h"');

/* eslint-disable no-var */

var AppHomeTest = class extends PolymerTest {
  /** @override */
  get browsePreload() {
    throw new Error('this is abstract and should be overriden by subclasses');
  }
};

var AppListAppTest = class extends AppHomeTest {
  /** @override */
  get browsePreload() {
    return 'chrome://apps/test_loader.html?module=app_home/app_list_test.js';
  }

  get featureList() {
    return {enabled: ['features::kDesktopPWAsAppHomePage']};
  }
};

TEST_F('AppListAppTest', 'All', function() {
  mocha.run();
});

File: chrome/test/data/webui/app_home/app_home_browsertest.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/test/scoped_feature_list.h"
#include "chrome/browser/ui/ui_features.h"
#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "content/public/test/browser_test.h"

class AppHomeTest : public WebUIMochaBrowserTest {
 protected:
  AppHomeTest() { set_test_loader_host(chrome::kChromeUIAppLauncherPageHost); }
};

IN_PROC_BROWSER_TEST_F(AppHomeTest, AppList) {
  RunTest("app_home/app_list_test.js", "mocha.run()");
}
